src/components/ChatLog.js

```diff
@@ -0,0 +1,43 @@
+import React from 'react';
+import './ChatLog.css';
+import PropTypes from 'prop-types';
+import ChatEntry from './ChatEntry.js'
+
+const ChatLog = (props) => {
+    if (!props.chatData) {
+        return null
+    }
+    const ChatLogEntries = props.chatData.map((entry) => {
+        return (
+            <div key={entry.id}>
+                <ChatEntry 
+                    id={entry.id} 
+                    sender={entry.sender} 
+                    timeStamp={entry.timeStamp} 
+                    body={entry.body} 
+                    liked={entry.liked}
+                    onUpdateChat={props.onUpdateChat}
+                    chatEntry={entry.sender===props.chatData[0].sender ? 'chat-entry local' : 'chat-entry remote'}
+                ></ChatEntry>
+            </div>
+        );
+    });
+    
+    return (
+        <div>
+            {ChatLogEntries}
+        </div>
+    );
+};
+
+ChatLog.propTypes = {
+    chat: PropTypes.arrayOf(PropTypes.shape({
+        id: PropTypes.number.isRequired,
+        senderData: PropTypes.string.isRequired,
+        bodyData: PropTypes.string.isRequired,
+        likedData: PropTypes.bool.isRequired
+    })),
+    onUpdateChat: PropTypes.func.isRequired
+};
```

Comment:
The PropType check here is just a little off. `chat` should be called `chatData` because that's what you named the prop you pass to ChatLog (see App.js line 41). We need to make sure that timeStamp is also included and that chatData has .isRequired too. 

Also, prefer to name the keys the same as they appear in messages.json so that senderData is just sender (like it appears in the data file.
```
ChatLog.propTypes = {
    chatData: PropTypes.arrayOf(
        PropTypes.shape({
            id: PropTypes.number.isRequired,
            sender: PropTypes.string.isRequired,
            timeStamp: PropTypes.string.isRequired,
            body: PropTypes.string.isRequired,
            liked: PropTypes.bool.isRequired
        })
    ).isRequired,
    onUpdateChat: PropTypes.func.isRequired
};